### PR TITLE
Add group creation, property setting, and additional layouts

### PR DESCRIPTION
**Pull request by mrjoshida**
Add commands for creating a group of anonymous sprites, setting group properties, and new layouts (circle and random).

### CODE REVIEW

File: src/p5.dance.js [hunk 1]
```diff
@@ -299,6 +299,14 @@ export default class DanceParty {
     return sprite;
   }
 
+  makeNewDanceSpriteGroup(n, costume, layout) {
```
Comment (joshlory) [inline src/p5.dance.js]: Should we put a cap on `n`?

File: src/p5.dance.js [hunk 2]
```diff
@@ -403,12 +423,17 @@ export default class DanceParty {
...
         sprite.y = 200;
       }
-    } else {
+    } else if (format === "column") {
       for (i=0; i<count; i++) {
         sprite = group[i];
         sprite.x = 200;
         sprite.y = (i+1) * (400 / (count + 1));
       }
+    } else if (format == "random") {
```
Comment (joshlory) [inline src/p5.dance.js]: Nit: `===`

File: src/p5.dance.js (hunk 1 above)
Comment (mrjoshida) [reply src/p5.dance.js]: I'm capping it in the block by providing a dropdown (currently set to 3-10)

Comment (joshlory) [reply src/p5.dance.js]: We probably want to also cap it here (or in Dance.js `nativeAPI. makeNewDanceSpriteGroup` call).  A user can hand-edit the XML to bypass any dropdown limits.

**Comment by joshlory** [comment]
image

Rad! `nativeAPI.makeNewDanceSpriteGroup(10, 'MOOSE', 'circle');`

File: src/p5.dance.js (hunk 1 above)
Comment (mrjoshida) [reply src/p5.dance.js]: Does it make more sense to cap the total number of sprites across the board instead. Even if we cap this block, you could just run it multiple times (which I would expect is a more common student workaround than editing XML)

Comment (mrjoshida) [reply src/p5.dance.js]: Or we let it be and see kids enjoy the Star Wars infinite droid effect :)

**Pull request merged**